bin/submitIndependentControlWorkflows.sh

```diff
@@ -0,0 +1,128 @@
+
+q=no
+a=no
+t=scanpy-galaxy
+u=no
+w=no
+m=5
+
+while getopts ":q:a:t:u:w:m:" o; do
+    case "${o}" in
+        q)
+            q=${OPTARG}
+            ;;
+        a)
+            a=${OPTARG}
+            ;;
+        t)
+            t=${OPTARG}
+            ;;
+        u)
+            u=${OPTARG}
+            ;;
+        w)
+            w=${OPTARG}
+            ;;
+        m)
+            m=${OPTARG}
+            ;;
+        *)
+            usage
+            exit 0
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+maxExpsToRun=$m
+
+workflow=scxa-control-workflow
+controlJobSuffix=${SCXA_ENV}_$workflow
+
+export SCXA_RESULTS=$SCXA_WORKFLOW_ROOT/results
+submissionMarker="$SCXA_RESULTS/.submitting"
+
+if [ -e  $submissionMarker ]; then
+    echo "Submission process ongoing"
+else
+    touch $submissionMarker
+fi
+
+currentJobs=$(bjobs -w | grep $controlJobSuffix)
+nJobsRunning=$(echo $currentJobs | wc -l)
+maxToSubmit=$((maxExpsToRun-nJobsRunning))
+
+if [ $nJobsRunning -ge $maxExpsToRun ]; then
+    echo "$nJobsRunning already running, submitting no more\n"
+    exit 0
+else
+    echo "Will sumbit a maximum of $maxToSubmit jobs"
+fi
+
+submitted=0
+
+# Fetch the Git SDRFs
+
+if [ ! -d 'metadata' ]; then
+    git clone $SCXA_METADATA_REPO metadata
+fi
+
+pushd metadata > /dev/null
+git pull > /dev/null
+popd > /dev/null
+
+# Submit workflows for any updatated metdata files. The Nextflow workflow
+# itself contains logic for not re-analaysing where it does not need to.
+
+export PATH=$SCXA_WORKFLOW_ROOT/workflow/scxa-control-workflow/bin:$PATH
+
+while read -r idfFile; do
+    idfFileName=$(basename $idfFile)
+    expId=$(echo $idfFileName | awk -F'.' '{print $1}')
+    sdrfFile=$(dirname $idfFile)/${expId}.sdrf.txt
+    
+    grep -P "$expId\\t" $SCXA_RESULTS/excluded.txt > /dev/null
+    if [ $? -eq 0 ]; then
+        experimentStatus='excluded'
+    else 
+        # Notes: checkExperimentStatus.sh is the same as that used by the
+        # Nextflow workflow, so the logic is kept consistent   
+        experimentStatus=$(checkExperimentStatus.sh $expId $sdrfFile $overwrite)
+    fi
+
+    echo "Experiment status for $expId: $experimentStatus"
+    currentlyRunning='false'
+
+    # Check if the jobs is already running
+
+    echo -e "$currentJobs" | grep "${expId}_$controlJobSuffix" > /dev/null
+    if [ $? -eq 0 ]; then
+        currentlyRunning='true'
+        echo "$expId is currently running"
+    fi
+
+    if [ "$currentlyRunning" = 'false' ] && [ $submitted -lt $maxToSubmit ] && [ "$experimentStatus" == 'new' ]; then
+
+        echo "Submitting $expId for re-analysis"
+        $SCXA_WORKFLOW_ROOT/workflow/scxa-control-workflow/bin/submitControlWorkflow.sh -e $expId -q $q -t $t -a $a -u $u -w $w 
+
+        currentJobs=$(bjobs -w | grep $controlJobSuffix)
+        submitted=$((submitted+1))
+
+    fi
+done <<< "$(ls $SCXA_WORKFLOW_ROOT/metadata/*/*/*.idf.txt)"
```

Comment:
I have seen odd behaviours in the past when using this mode of feeding loops in bash, I always end up changing it for something that simply iterates of the list of strings, like `for file in $(ls ...); do cat $file`. I think that each loop gets its own process and that has its own consequences/particularities with dealing with vars and so on.

Reply:
Yep, while read has that issue, but the here string resolves it. I had reasons for adopting this pattern over simple for loops in the past, but I forget what it is now. Anyway- this does work.